static/js/message_edit.js

```diff
@@ -562,7 +567,15 @@ export function start_topic_edit(recipient_row) {
     if (topic === compose.empty_topic_placeholder()) {
         topic = "";
     }
-    form.find(".inline_topic_edit").val(topic).trigger("select").trigger("focus");
+    const inline_topic_edit_input = form.find(".inline_topic_edit");
+    inline_topic_edit_input.val(topic).trigger("select").trigger("focus");
+    inline_topic_edit_input.on("keydown", (e) => {
+        // prevent the form from submitting, so user can select an item from typeahead.
+        if (e.code === "Enter") {
+            e.stopPropagation();
+        }
+    });
```

Comment:
We mentioned [here](https://github.com/zulip/zulip/pull/14033#issuecomment-594245087) that we'd prefer to do this on the typeahead side, so I think we'd like to have commit as is https://github.com/zulip/zulip/pull/14033/commits/ed21080c6ba3deae0f21ae22ceee5eaf7a941cbc

Reply:
Oh so should I close this pr. Or may  include `[ed21080](https://github.com/zulip/zulip/commit/ed21080c6ba3deae0f21ae22ceee5eaf7a941cbc)` in this pr. Also sorry for late reply on this.